homeassistant/components/smarttub/binary_sensor.py

```diff
@@ -0,0 +1,36 @@
+"""Platform for binary sensor integration."""
+import logging
+
+from homeassistant.components.binary_sensor import STATE_ON, BinarySensorEntity
+
+from .const import DOMAIN, SMARTTUB_CONTROLLER
+from .sensor import SmartTubSensor
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up binary sensor entities for the binary sensors in the tub."""
+
+    controller = hass.data[DOMAIN][entry.entry_id][SMARTTUB_CONTROLLER]
+
+    entities = []
+    for spa in controller.spas:
+        entities.append(
+            SmartTubOnline(controller.coordinator, spa),
+        )
```

Comment:
This would be shorter as a list comprehension but if you add more later, it probably doesn't make sense to change it.

```suggestion
    entities = [SmartTubOnline(controller.coordinator, spa) for spa in controller.spas]
```